### PR TITLE
Stop and raise an exception if loss is nan

### PR DESCRIPTION
**Pull request by sovrasov**
Also keys of the losses summary dict were put in order

### CODE REVIEW

File: torchreid/engine/engine.py [hunk 1]
```diff
@@ -510,6 +511,8 @@ def train(self, print_freq=10, fixbase_epoch=0, open_layers=None, lr_finder=Fals
 
             loss_summary, avg_acc = self.forward_backward(data)
             batch_time.update(time.time() - end)
+            if math.isnan(loss_summary[self.get_model_names()[0]]):
```
Comment (kprokofi) [inline torchreid/engine/engine.py]: We can check if the loss is inf either

File: torchreid/engine/image/am_softmax.py [hunk 2]
```diff
@@ -172,8 +172,6 @@ def forward_backward(self, data):
                     self.scaler.update()
 
         # record losses
-        if mutual_learning:
```
Comment (kprokofi) [inline torchreid/engine/image/am_softmax.py:175]: why did you remove mutual loss dumping?

Comment (sovrasov) [reply torchreid/engine/image/am_softmax.py:175]: `i` variable as well as `mutual_loss` are used outside of the loop context here, this shouldn't happen at all. Also this value was already dumped in the loop.

File: torchreid/engine/engine.py (hunk 1 above)
Comment (sovrasov) [reply torchreid/engine/engine.py]: Added this check as well

**Review by kprokofi — APPROVED** [review]

**Comment by kprokofi** [comment]
@sovrasov , please fix the pylint errors and and you can merge

**Comment by sovrasov** [comment]
The failure is not related to this PR. #158 contains a fix for that
